src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -214,3 +227,121 @@ def process_loaded_scg_route(self):
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            stringErr = str(e)
+            if "JavaComponentsNotAllowedForSubscription" in stringErr:
+                raise CLIInternalError("Java Components operations are not allowed for the subscription, please use 'az feature register --namespace  Microsoft.App --name JavaComponentsPreview' to register this feature.")
+
+            handle_raw_exception(e)
+
+    # need handle already set configurations
+    def set_up_set_configurations(self):
+        if self.get_argument_set_configurations() is not None:
+            configuration_list = self.parse_configurations(self.get_argument_set_configurations())
+            self.java_component_def["properties"]["configurations"] = configuration_list
+
+    def construct_payload(self):
+        self.set_up_component_type()
+        self.set_up_service_bindings()
+        self.set_up_unbind_service_bindings()
+        self.set_up_gateway_route()
+        self.set_up_replicas()
+        self.validate_configurations()
+        # --configuration
+        self.set_up_configurations()
+        # --set-configuration
+        self.set_up_set_configurations()
+
+
+class JavaComponentUpdateDecorator(BaseJavaComponentDecorator):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        current_java_component_def = self.show()
+        if "configurations" not in current_java_component_def["properties"] or not current_java_component_def["properties"]["configurations"]:
+            self.java_component_def["properties"]["configurations"] = []
+        else:
+            self.java_component_def["properties"]["configurations"] = current_java_component_def["properties"]["configurations"]
+
+    def update(self):
+        try:
+            return self.client.update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def add_or_update_configurations(self, existing_configurations, new_configurations):
+        for new_configuration in new_configurations:
+            if "value" not in new_configuration:
+                raise ValidationError("Value must be provided for configuration {}.".format(new_configuration["propertyName"]))
+
+            # Check if updating existing configuration
+            is_existing = False
+            for existing_configuration in existing_configurations:
+                if existing_configuration["propertyName"].lower() == new_configuration["propertyName"].lower():
```

Comment:
They are case-sensitive, why you `lower()` them to do the comparison? We should not change the customer input to compare.